### PR TITLE
fix(condo): DOMA-3718 added ticket deadline from resident mobile app

### PR DESCRIPTION
**Pull request by Alllex202**
Problem: tickets from resident app dont have deadline

### CODE REVIEW

**Review by pahaz — CHANGES_REQUESTED** [review]

File: apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js [hunk 1]
```diff
@@ -13,6 +14,8 @@ const { COMPLETED_STATUS_TYPE, NEW_OR_REOPENED_STATUS_TYPE } = require('@condo/d
 const { FLAT_UNIT_TYPE, SECTION_SECTION_TYPE, PARKING_UNIT_TYPE, PARKING_SECTION_TYPE } = require('@condo/domains/property/constants/common')
 const { DEFERRED_STATUS_TYPE, DEFAULT_DEFERRED_DAYS } = require('@condo/domains/ticket/constants')
 
+dayjs.extend(duration)
```
Comment (pahaz) [inline apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js:17]: why you really needed to use dayjs inside the seter side logic? you can use Date.

File: apps/condo/domains/ticket/schema/Ticket.test.js [hunk 2]
```diff
@@ -65,6 +68,8 @@ const {
 const { WRONG_VALUE } = require('@app/condo/domains/common/constants/errors')
 const { get } = require('lodash')
 
+dayjs.extend(duration)
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.test.js:71]: You should do it once inside index.js. And don't move it from one file to another

File: .github/workflows/nodejs.condo.test.yml [hunk 3]
```diff
@@ -77,18 +77,18 @@ jobs:
           sleep 3
 
           # TESTS
-          yarn workspace @app/condo test --testTimeout=15000 --runInBand --forceExit --silent=false --verbose --bail --testPathPattern '/domains/(organization|user)/schema/(.*)[.]test.js$' 2>&1 > condo.1.test.organization-user.log
+          #yarn workspace @app/condo test --testTimeout=15000 --runInBand --forceExit --silent=false --verbose --bail --testPathPattern '/domains/(organization|user)/schema/(.*)[.]test.js$' 2>&1 > condo.1.test.organization-user.log
```
[the remark this answers is not in the document]
Comment (Alllex202) [reply .github/workflows/nodejs.condo.test.yml:80]: WIP

Gradually adding tests, I check at what point they will stop passing
I talked about it at stand-up

File: apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js (hunk 1 above)
Comment (Alllex202) [reply apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js:17]: dayjs can work with duration

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 4]
```diff
@@ -424,6 +425,7 @@ const Ticket = new GQLListSchema('Ticket', {
                 overrideTicketFieldsForResidentUserType(context, resolvedData)
                 await setSectionAndFloorFieldsByDataFromPropertyMap(context, resolvedData)
                 setClientNamePhoneEmailFieldsByDataFromUser(get(context, ['req', 'user']), resolvedData)
+                await setDeadline(resolvedData)
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:428]: it's better to write something like:

if (!get(resolvedData, 'deadline') && !get(existingItem, 'deadline')) awawit setDeadline(...)

Probably, someone can use our API and can forgot to set this field

File: apps/condo/index.js [hunk 5]
```diff
@@ -35,6 +37,8 @@ const { FeaturesMiddleware } = require('@condo/featureflags/FeaturesMiddleware')
 const packageJson = require('@app/condo/package.json')
 const { featureToggleManager } = require('@condo/featureflags/featureToggleManager')
 
+dayjs.extend(duration)
```
Comment (pahaz) [inline apps/condo/index.js:40]: 👍

File: apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js [hunk 6]
```diff
@@ -233,6 +235,51 @@ async function connectContactToTicket (context, resolvedData, existingItem) {
...
+    const ticketSetting = await getByCondition('TicketOrganizationSetting', {
+        organization: { id: organizationId },
+    })
+    const isPaid = get(newItem, 'isPaid')
+    const isEmergency = get(newItem, 'isEmergency')
+    const isWarranty = get(newItem, 'isWarranty')
+    const deadlineDurationAsMs = getDeadlineDurationAsMs(ticketSetting, { isPaid, isEmergency, isWarranty })
+    const defaultDeadline = dayjs().add(deadlineDurationAsMs, 'ms').toISOString()
+
+    if (operation === 'create' || operation === 'update' && !isUndefined(resolveDeadline)) {
```
Comment (Alllex202) [inline apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js:271]: Check attribute changes (paid/warranty/emergency) if update

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 7]
```diff
@@ -424,6 +428,7 @@ const Ticket = new GQLListSchema('Ticket', {
...
```
Comment (akarjakin) [inline apps/condo/domains/ticket/schema/Ticket.js:431]: Please don't mutate function's arguments under the hood. Try to make it more obvious

Comment (Alllex202) [reply apps/condo/domains/ticket/schema/Ticket.js:431]: There are several more functions like this

**Review by pahaz — APPROVED** [review]

**Review by nomerdvadcatpyat — APPROVED** [review]

**Pull request merged**